test/powershell/Modules/Microsoft.PowerShell.Core/Enter-PSHostProcess.Tests.ps1

```diff
@@ -1,134 +1,163 @@
+                    Start-Sleep -Seconds 30 | Out-Null
                 }
-                $powershell = Start-Process @params
             }
 
+            $pwshId = Wait-JobPid $pwshJob
         }
 
-        AfterAll {
-            $pwsh | Stop-Process
-
-            if ($IsWindows) {
-                $powershell | Stop-Process
-            }
+        AfterEach {
+            $pwshJob | Stop-Job -PassThru | Remove-Job
         }
 
-        It "Can enter and exit another PSHost" -Pending:$true {
-            Wait-UntilTrue { (Get-PSHostProcessInfo -Id $pwsh.Id) -ne $null }
+        It "Can enter, exit, and re-enter another PSHost" {
+            Wait-UntilTrue { [bool](Get-PSHostProcessInfo -Id $pwshId) }
 
-            "Enter-PSHostProcess -Id $($pwsh.Id) -ErrorAction Stop
-            `$pid
-            Exit-PSHostProcess" | pwsh -c - | Should -Be $pwsh.Id
+            # This will enter and exit another process
+            Invoke-PSHostProcessScript -ArgumentString "-Id $pwshId" -Id $pwshId |
+                Should -BeTrue -Because "The script was able to enter another process and grab the pid of '$pwshId'."
+
+            # Re-enter and exit the other process
+            Invoke-PSHostProcessScript -ArgumentString "-Id $pwshId" -Id $pwshId |
+                Should -BeTrue -Because "The script was able to enter another process and grab the pid of '$pwshId'."
         }
 
-        It "Can enter and exit another Windows PowerShell PSHost" -Skip:(!$IsWindows) {
-            Wait-UntilTrue { (Get-PSHostProcessInfo -Id $powershell.Id) -ne $null }
+        It "Can enter, exit, and re-enter another Windows PowerShell PSHost" -Skip:(!$IsWindows) {
+            # Start a Windows PowerShell job where the first thing it does is return $pid. After that, spin forever.
+            # We will use this job as the target process for Enter-PSHostProcess
+            $powershellJob = Start-Job -PSVersion 5.1 {
+                $pid
+                while ($true) {
+                    Start-Sleep -Seconds 30 | Out-Null
+                }
+            }
+
+            $powershellId = Wait-JobPid $powershellJob
+
+            try {
+                Wait-UntilTrue { [bool](Get-PSHostProcessInfo -Id $powershellId) }
+
+                # This will enter and exit another process
+                Invoke-PSHostProcessScript -ArgumentString "-Id $powershellId" -Id $powershellId |
+                    Should -BeTrue -Because "The script was able to enter another process and grab the pid of '$powershellId'."
 
-            "Enter-PSHostProcess -Id $($powershell.Id) -ErrorAction Stop
-            `$pid
-            Exit-PSHostProcess" | pwsh -c - | Should -Be $powershell.Id
+                # Re-enter and exit the other process
+                Invoke-PSHostProcessScript -ArgumentString "-Id $powershellId" -Id $powershellId |
+                    Should -BeTrue -Because "The script was able to enter another process and grab the pid of '$powershellId'."
```

Comment:
```suggestion
                    Should -BeTrue -Because "The script was able to re-enter another process and grab the pid of '$powershellId'."
```

Reply:
fixed